src/System.Diagnostics.EventLog/src/System/Diagnostics/Reader/EventLogException.cs

```diff
@@ -61,54 +63,78 @@ public override string Message
         }
 
         private int _errorCode;
+
+        protected EventLogException(SerializationInfo serializationInfo, StreamingContext streamingContext)
+            : base(serializationInfo, streamingContext)
+        {
+            _errorCode = serializationInfo.GetInt32("errorCode");
+        }
+
+        public override void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            if (info == null)
+            {
+                throw new ArgumentNullException(nameof(info));
+            }
+            info.AddValue("errorCode", _errorCode);
+            base.GetObjectData(info, context);
```

Comment:
The base call should be at the beginning and you don't have to check if info is null as that's handled in the base class. An example: https://source.dot.net/#System.Private.CoreLib/shared/System/ArgumentException.cs,73